controllers/propagator/placement_change_reconclier.go

```diff
@@ -0,0 +1,190 @@
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"open-cluster-management.io/governance-policy-propagator/controllers/common"
+)
+
+//+kubebuilder:rbac:groups=policy.open-cluster-management.io,resources=placementbindings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.open-cluster-management.io,resources=placementrules,verbs=get;list;watch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PlacementChangeReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	placementBindingMapper := func(ctx context.Context, obj client.Object) []reconcile.Request {
+		//nolint:forcetypeassert
+		pb := obj.(*policiesv1.PlacementBinding)
+
+		log := log.WithValues("placementBindingName", pb.GetName(), "namespace", pb.GetNamespace())
+		log.V(2).Info("Reconcile request for a PlacementBinding")
+
+		return common.GetPoliciesInPlacementBinding(ctx, r.Client, pb)
+	}
+
+	// only reconcile when the pb contains a policy or a policyset as a subject
+	pbPredicateFuncs := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			//nolint:forcetypeassert
+			pbObjNew := e.ObjectNew.(*policiesv1.PlacementBinding)
+			//nolint:forcetypeassert
+			pbObjOld := e.ObjectOld.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObjNew) || common.IsForPolicyOrPolicySet(pbObjOld)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			//nolint:forcetypeassert
+			pbObj := e.Object.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObj)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			//nolint:forcetypeassert
+			pbObj := e.Object.(*policiesv1.PlacementBinding)
+
+			return common.IsForPolicyOrPolicySet(pbObj)
+		},
+	}
+
+	// placementRuleMapper maps from a PlacementRule to the PlacementBindings associated with it
+	placementRuleMapper := func(ctx context.Context, object client.Object) []reconcile.Request {
+		log := log.WithValues("placementRuleName", object.GetName(), "namespace", object.GetNamespace())
+		log.V(2).Info("Reconcile Request for PlacementRule")
+
+		pbList := &policiesv1.PlacementBindingList{}
+		lopts := &client.ListOptions{Namespace: object.GetNamespace()}
+
+		opts := client.MatchingFields{"placementRef.name": object.GetName()}
+		opts.ApplyToList(lopts)
+
+		if err := r.List(ctx, pbList, lopts); err != nil {
+			return nil
+		}
+
+		var result []reconcile.Request
+		// loop through pbs find the matching ones
+		for _, pb := range pbList.Items {
+			match := pb.PlacementRef.APIGroup == appsv1.SchemeGroupVersion.Group &&
+				pb.PlacementRef.Kind == "PlacementRule" &&
+				pb.PlacementRef.Name == object.GetName()
```

Comment:
No harm in having this but doesn't the list option from before ensure this?

Reply:
Yes, but because I wasn't 100% sure if the list from the cache would honor that filter, I wanted to keep it here.